File: DailyPractice/Week28/fairDistribution.py
from os import path
import sys 

class Solution:
    def solve(self):
        n , k = list(map(int , input().split()))
        lst = list(map(int, input().split()))
        lst.sort()
        mini = float("inf")
        for i in range(n - k + 1):
            mini = min(mini , lst[i + k - 1] - lst[i])
        print(mini)

s = Solution()
if path.exists('input.txt'):
    sys.stdin = open('input.txt', "r")
    sys.stdout = open('output.txt' , "w")
t = int(input())
while t != 0:
    s.solve()
    t -= 1

File: DailyPractice/Week28/fairDistribution.cpp
#include<bits/stdc++.h>
#define ll             long long int
#define F              first
#define S              second
#define pb             push_back
#define vi             vector <int>
#define pii            pair <int, int>
#define rep(n) for (int i = 0; i < n; i++)
using namespace std;
class Solution{
    public:
    void solve(){
        int n, k;
        cin >> n >> k;
        vi arr(n);
        rep(n){
            cin >> arr[i];
        }
        sort(begin(arr), end(arr));
        int mini = INT_MAX;
        for (int i = 0; i < n - k + 1; i++){
            mini = min(mini, arr[i + k - 1] - arr[i]);
        }
        cout << mini;
    }
};


int main(){

#ifndef ONLINE_JUDGE
    freopen("input.txt","r",stdin);
    freopen("output.txt","w",stdout);
#endif
    Solution s;
    int t;
    cin >> t;
    while(t--){
        s.solve();
        cout<<'\n';
    }
}